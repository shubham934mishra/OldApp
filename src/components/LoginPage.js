
//@flow
import React, { Component } from 'react';
import {
    View,
} from 'react-native';
import connect from 'react-redux/lib/connect/connect';
import { getLoginData } from '../actions/LoginAction';
import { NAVIGATION_DASHBOARD } from '../navigations/routes';
import { styles } from '../utilities/style';
import { CustomTextInput } from '../globalComponents/CustomTextInput';
import { AppButton } from '../globalComponents/CustomButton';

const mapStateToProps = (state) => {
    const dataSource = state.LoginReducer.data;
    return {
        dataSource
    };
};

const mapDispatchToProps = (dispatch) => ({
    callService: (loginObj) => dispatch(getLoginData(loginObj)),
});


class LoginPage extends Component {

    state = {
        userName: '',
        password: '',
        device: '',
    };

    onGuestPress = () => {
        const loginObj = {
            userName: this.state.userName,
            password: this.state.password,
            device: this.state.device
        };
        this.props.callService(loginObj);
        this.props.navigation.navigate(NAVIGATION_DASHBOARD);
    }

    render() {
        return (
            <View style={styles.container}>

                <CustomTextInput
                    testID={'TextInput'}
                    label='User Name'
                    style={styles.TextInput}
                    ipOnChangeText={userName => this.setState({ userName })}
                />

                <CustomTextInput
                    testID={'TextInputPass'}
                    style={styles.TextInput}
                    label="Password"
                    ipOnChangeText={password => this.setState({ password })}
                />

                <AppButton
                    testID='loginButton'
                    stylee={styles.loginBtn}
                    onPress={this.onGuestPress}
                    title='LOGIN'
                />
            </View>
        );
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
