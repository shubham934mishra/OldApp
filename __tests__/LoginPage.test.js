/* eslint-disable no-undef */

import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import thunk from 'redux-thunk';
import LoginPage from '../src/components/LoginPage';


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ LoginReducer: { data: 'shubham' } });

const baseProps = {
  navigation: {
    navigate: jest.fn()
  },
  dataSource: {},
  callService: jest.fn(),
  store: jest.fn(),
  dispatch: jest.fn()
};


test('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}>
    <LoginPage {...baseProps} />
  </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

//Demo test
describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  });
});


test('Testing in Login Page', async () => {
  const { getByTestId, queryByText } = render(
    <Provider store={store}>
      <LoginPage {...baseProps} />
    </Provider>
  );

  // Test For TextInput in Login Screen
  const textInput = getByTestId('TextInput');
  const TextInputPass = getByTestId('TextInputPass');
  fireEvent.changeText(textInput, 'Shubham mishra: checked');
  fireEvent.changeText(TextInputPass, 'shubham: Pass check');


  // Test For Buttons in Login Screen 
  const navigateToButtons = getByTestId('loginButton');
  fireEvent.press(navigateToButtons);
  const loginKey = await queryByText('Shubham : login');
  expect(loginKey).toBeDefined();
  expect(baseProps.navigation.navigate)
    .toHaveBeenCalledWith('Dashboard');
}
);
