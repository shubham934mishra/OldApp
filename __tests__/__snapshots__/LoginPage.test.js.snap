// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#7a42f4",
        "borderWidth": 1,
        "height": 40,
        "margin": 15,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="User Name"
      placeholderTextColor="#003f5c"
      rejectResponderTermination={true}
      testID="TextInput"
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "borderColor": "#7a42f4",
        "borderWidth": 1,
        "height": 40,
        "margin": 15,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#003f5c"
      rejectResponderTermination={true}
      testID="TextInputPass"
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#7a42f4",
        "height": 40,
        "margin": 15,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="loginButton"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
        }
      }
    >
      LOGIN
    </Text>
  </View>
</View>
`;
